### PR TITLE
use consistent product name `WooCommerce Blocks`

### LINKED ISSUE #3066: Rename WooCommerce Product Blocks to WooCommerce Blocks
When the project was first created, the focus was on product blocks, but in the time since this project ecompasses (and will encompass) more than just product blocks. So the purpose of this issue is to rename the project to a more simplified "WooCommerce Blocks".

What does this require:

* [x] Changing the name of the plugin wherever it is referenced in code and documents.
* [x] Updating name of plugin in WooCommerce.com.
* [ ] Changing the slug for the github repo (🚨This could be dangerous and we'll need to make sure it won't break anything before doing).
* [x] Revisit [this paragraph in the README.md for the repo](https://github.com/woocommerce/woocommerce-gutenberg-products-block#vision-for-the-feature) and update it to reflect the current status of the project.
* We **can't** change the slug for the wp.org repo (just including this to head off potential questions).

### PR DESCRIPTION
**Pull request by haszari**
Fixes #3066

This PR refreshes various references to `WooCommerce Gutenberg Products Block` (product name in older iterations) to the correct product name `WooCommerce Blocks`. 

Also refreshes the vision sentence in the readme - cc @garymurray - let's get this right, I've done an initial draft. We might consider using this for the org tagline too.

Screen Shot 2020-12-18 at 9 56 36 AM

Note there are still some things we could tidy up in WCCOM and docs, to really make the messaging consistent around why this plugin exists and when to use it (and the fact that most of the blocks are in core). 

Also I haven't changed any slugs. We could consider changing the WCCOM url slug and the github repo slug, I suggest doing these individually as follow up (with care!).

### How to test the changes in this Pull Request:

1. Review the docs changes.

### CODE REVIEW

**Review by budzanowski — APPROVED** [review]

**Comment by haszari** [comment]
Merging! This mainly affects repo docs so is low risk. At the same time there may be some follow up to make the mission/vision consistent everywhere. 

TLDR For when people come back from AFK (@garymurray @nerrad): in this PR I re-drafted the readme "vision" sentence to (see below), keen for feedback on this, and eventually update in other places (WCCOM, WPORG) as follow up.

`WooCommerce Blocks are the easiest, most flexible way to build your store user interface and showcase your products.`
